### PR TITLE
#175636842 registration validation errors

### PR DESCRIPTION
**Pull request by geraudirad**
#### What does this PR do?
Validates the signup inputs and send error message if it is invalid
#### Description of Task to be completed?
- Added Joi package for validation
- Defined validation schema
- Added signup validation middleware
#### How should this be manually tested?
Send Request to localhost:3000/api/v1/user/signup with user information
#### What are the relevant pivotal tracker stories?
[175636842](https://www.pivotaltracker.com/story/show/175636842)
#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by Duncanwar — APPROVED** [review]

File: src/routes/api/users.route.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ const router = express.Router();
 const { signup } = UserControllers;
 const { loginCallback } = AuthControllers;
 const { checkEmailExist } = authMiddleware;
+import {UserValidation} from "../../middlewares/user"
+
+
+
+//const { signupValidation } = UserValidation;
```
Comment (kipropbrian) [inline src/routes/api/users.route.js]: Remove these unnecessary spaces and comments

Comment (geraudirad) [reply src/routes/api/users.route.js]: alright, let me do it

**Review by Muhire-Josue — APPROVED** [review]
LGTM

**Comment by Muhire-Josue** [comment]
Please rebase, then I can merge it

**Pull request merged**